lib/Onyx.js

```diff
@@ -492,7 +476,10 @@ function remove(key) {
  */
 function evictStorageAndRetry(error, ionMethod, ...args) {
     // Find the first key that we can remove that has no subscribers in our blocklist
-    const keyForRemoval = _.find(recentlyAccessedKeys, key => !evictionBlocklist[key]);
+    const keyForRemoval = _.find(
+        recentlyAccessedKeys,
+        key => !evictionBlocklist[key] && isSafeEvictionKey(key),
```

Comment:
This check `isSafeEvictionKey` was moved here to allow to use `recentlyAccessedKeys` to actually put all recently accessed keys to that list